flask/templating.py

```diff
@@ -101,7 +101,19 @@ def list_templates(self):
 
 
 def _render(template, context, app):
-    """Renders the template and fires the signal"""
+    """Renders the template and fires signals.
+
+    It is possible to override render template in the before_render_template signal.
+    It can be done only if exactly one receiver and it return not None result."""
+
+    brt_resp = before_render_template.send(app, template=template, context=context)
+    
+    if len(brt_resp) == 1:
```

Comment:
I think this can be simplified a bit:

```python
if len(brt_resp) == 1 and brt_resp[0][1] is not None:
    return brt_resp[0][1]
```

Signals always return `func, func_return_value` tuples, so you don't need to check that you have two elements.

Another possibility would be this:

```python
overrides = [rv for _, rv in brt_resp if rv is not None]
if len(overrides) == 1:
    return overrides[0]
elif len(overrides) > 1:
    raise RuntimeError('More than one before_render_template signal '
                       'returned data')
```